src/ddscxx/src/org/eclipse/cyclonedds/core/EntityDelegate.cpp

```diff
@@ -62,6 +62,12 @@ org::eclipse::cyclonedds::core::EntityDelegate::~EntityDelegate()
 
 void org::eclipse::cyclonedds::core::EntityDelegate::enable()
 {
+  if (DDS_BUILTIN_TOPIC_DCPSPARTICIPANT == this->ddsc_entity
+   || DDS_BUILTIN_TOPIC_DCPSTOPIC == this->ddsc_entity
+   || DDS_BUILTIN_TOPIC_DCPSSUBSCRIPTION == this->ddsc_entity
+   || DDS_BUILTIN_TOPIC_DCPSPUBLICATION == this->ddsc_entity)
+    return;
```

Comment:
Why do I prematurely return if the Entity I want to enable belongs to a builtin topic? I guess the builtin topics are already enabled by default, but shouldn't the enable() function be idempotent according to the spec? Why the need for an intercept here?